modules/caddyhttp/server.go

```diff
@@ -146,6 +146,56 @@ type Server struct {
 
 	shutdownAt   time.Time
 	shutdownAtMu *sync.RWMutex
+
+	// configuring http.Server
+	connStates   []func(net.Conn, http.ConnState)
+	connContexts []func(ctx context.Context, c net.Conn) context.Context
+	onShutdowns  []func()
+}
+
+// register ConnState callbacks
+func (s *Server) RegisterConnState(f func(net.Conn, http.ConnState)) {
+	s.connStates = append(s.connStates, f)
+}
+
+// register ConnContext callbacks
+func (s *Server) RegisterConnContext(f func(ctx context.Context, c net.Conn) context.Context) {
+	s.connContexts = append(s.connContexts, f)
+}
+
+// register OnShutdown callbacks
+func (s *Server) RegisterOnShutdown(f func()) {
+	s.onShutdowns = append(s.onShutdowns, f)
+}
+
+// apply callbacks to server
+func (s *Server) ConfigureServer(server *http.Server) {
```

Comment:
Does this have to be exported? It should only be configured within this package, right?

Reply:
Yes, it should be unexported.